src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/UnconfiguredProjectContextProvider.cs

```diff
@@ -164,11 +171,71 @@ private ProjectData GetProjectData()
+                var activeProjectConfiguration = _commonServices.ActiveConfiguredProject.ProjectConfiguration;
+
+                var innerProjectContextsBuilder = ImmutableDictionary.CreateBuilder<string, IWorkspaceProjectContext>();
+                IConfiguredProjectHostObject activeIntellisenseProjectHostObject = null;
+                foreach (var kvp in configuredProjectsMap)
+                {
+                    var targetFramework = kvp.Key;
+                    var configuredProject = kvp.Value;
+
+                    // Get the target path for the configured project.
+                    var projectProperties = configuredProject.Services.ExportProvider.GetExportedValue<ProjectProperties>();
+                    var configurationGeneralProperties = await projectProperties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(true);
+                    targetPath = (string)await configurationGeneralProperties.TargetPath.GetValueAsync().ConfigureAwait(true);
+                    targetPath = NormalizeTargetPath(targetPath, projectData);
+                    var displayName = GetDisplayName(configuredProject, projectData, targetFramework);
+                    IConfiguredProjectHostObject configuredProjectHostObject = _projectHostProvider.GetConfiguredProjectHostObject(unconfiguredProjectHostObject, displayName);
+
+                    // TODO: https://github.com/dotnet/roslyn-project-system/issues/353
+                    await _commonServices.ThreadingService.SwitchToUIThread();
+                    var workspaceProjectContext = _contextFactory.Value.CreateProjectContext(languageName, displayName, projectData.FullPath, projectGuid, configuredProjectHostObject, targetPath);
+                    innerProjectContextsBuilder.Add(targetFramework, workspaceProjectContext);
+
+                    if (activeIntellisenseProjectHostObject == null && configuredProject.ProjectConfiguration.Equals(activeProjectConfiguration))
+                    {
+                        activeIntellisenseProjectHostObject = configuredProjectHostObject;
+                    }
+                }
+
+                unconfiguredProjectHostObject.ActiveIntellisenseProjectHostObject = activeIntellisenseProjectHostObject;
 
-                return _contextFactory.Value.CreateProjectContext(languageName, projectData.DisplayName, projectData.FullPath, projectGuid, _commonServices.Project.Services.HostObject, targetPath);
+                var activeTargetFramework = activeProjectConfiguration.Dimensions[TargetFrameworkProjectConfigurationDimensionProvider.TargetFrameworkPropertyName];
+                return new AggregateWorkspaceProjectContext(innerProjectContextsBuilder.ToImmutable(), activeTargetFramework, unconfiguredProjectHostObject);
             });
         }
 
+        private static string NormalizeTargetPath(string targetPath, ProjectData projectData)
```

Comment:
Agreed. I have created https://github.com/dotnet/roslyn/issues/14520 to track this work on Roslyn side and https://github.com/dotnet/roslyn-project-system/issues/607 to track remove this workaround on our side once the roslyn issue is fixed.